### ISSUE TITLE
Feature request: Custom Tools can be provided with the cursor position

### ISSUE THREAD

**Issue by tridian-tn** [zim-desktop-wiki/zim-desktop-wiki#1607]
When I want to edit the source of the Zim page (via Tools->Edit Source), the page opens in the configured default text editor, but with the cursor at the top of the file. At this point, I have to page down to the point where I want to edit - e.g. often to find a table to make manual edits. This delay isn't trivial when I'm working on a page over a day.

I'm not suggesting esoteric edits to the Edit Source function, it would be less intrusive to have a command line parameter for Custom Tools whereby the cursor position in the source text can be provided to an external tool. For example, %s and a potential cursor position %c could be provided to an editor that supports that (e.g. Notepad3 on Windows). If the current focus is on a plugin (e.g. the table editor) then somewhere nearby is close enough.

Summary of request: Add a new % parameter to Custom Tools execution (and the Edit Custom Tools window), that provides the current cursor position in the raw source file. 

Many thanks for considering!

**Comment by jaap-karssenberg** [comment]
Not so easy as it sounds, since the cursor position in the zim editor is
related to the formatted text. What you want is the corresponding location
in the wiki source. Currently there is no logic in the code base to figure
that out.

Probably the solution would be to have a representation of the cursor (or a
more generic "position marker" with a name attached) in the parse tree and
let the dumper class of the formatter track the offset of these markers
when generating the output. Then still some wiring is needed to report back
these positions such that the custom tool could use them.

**Comment by tuchang** [comment]
Not so easy but `good first issue` label?

**Comment by jaap-karssenberg** [comment]
Yes, it is a good first issue because it allows understanding of how
serialization works without impacting other subsystems
